src/sentry/api/bases/organization_events.py

```diff
@@ -22,12 +21,16 @@ class Direction(object):
 
 class OrganizationEventsEndpointBase(OrganizationEndpoint):
     def get_snuba_query_args(self, request, organization, params):
+        snuba_args = {}
         query = request.GET.get("query")
         try:
-            snuba_args = get_snuba_query_args(query=query, params=params)
+            filter = get_filter(query, params)
         except InvalidSearchQuery as exc:
             raise OrganizationEventsError(exc.message)
 
+        for key in ("start", "end", "conditions", "filter_keys"):
+            snuba_args[key] = getattr(filter, key)
```

Comment:
Is this going to be permanent solution ?
Using getattr to access filter attributes will make it hard to refactor Filter because we will have to find this usage.
There are cleaner solutions like:

```
class Filter:
    ....
    def get_attributes(self, attrs: Sequence[str]):
         # do the same and return a dictionary.
        # This way the logic is coupled with Filter class and it is hard to miss when refactoring it
```

another is simply accessing the attributes one by one:
```
snuba_args = {
         "start": filter.start,
          "end": filter.end,
          "filter_keys": filter.filter_keys,
}
```

It does not look sensibly worse than getattr but it makes refactoring a lot easier

Reply:
Updated to accessing one by one, since that's how we dealt with this in other places here.